lib/ai/model/messages.go

```diff
@@ -66,6 +66,11 @@ type TokensUsed struct {
 	Completion int
 }
 
+// UsedTokens returns the number of tokens used during a single invocation of the agent.
+func (t *TokensUsed) UsedTokens() *TokensUsed {
```

Comment:
What does this even do? By what I gather, this interface only needs to be implemented for the possible return types for Complete() so this seems useless.

Reply:
Yes I get that but the message here is never a `TokensUsed`. I understand the method impl on Message and CompleteMessage, though.

Reply:
Implemented by embedding. That was my goal.

Reply:
Got it